easybuild/easyconfigs/j/jupyter-server-proxy/jupyter-server-proxy-3.2.2-GCCcore-11.3.0.eb

```diff
@@ -0,0 +1,41 @@
+easyblock = 'PythonBundle'
+
+name = 'jupyter-server-proxy'
+version = '3.2.2'
+
+homepage = 'https://github.com/jupyterhub/jupyter-server-proxy'
+description = """Jupyter Server Proxy lets you run arbitrary external processes
+(such as RStudio, Shiny Server, Syncthing, PostgreSQL, Code Server, etc)
+alongside your notebook server and provide authenticated web access to them    
+using a path like /rstudio next to others like /lab. Alongside the python
+package that provides the main functionality, the JupyterLab extension
+(@jupyterlab/server-proxy) provides buttons in the JupyterLab launcher window
+to get to RStudio for example."""
+
+toolchain = {'name': 'GCCcore', 'version': '11.3.0'}
+
+builddependencies = [
+    ('binutils', '2.38'),
+]
+
+dependencies = [
+    ('Python', '3.10.4'),
+    ('JupyterHub', '3.0.0'),
```

Comment:
`jupyter-server-proxy` was recently merged in #17270 including this change

Reply:
I wouldn't call JupyterLab a `lower` dependency. In fact, we only use `jupyter-server-proxy` in combination with `JupyterHub` and _regular_ `Jupyter` notebooks. So... no Lab involved in our stack. We use the proxy for connecting to e.g. TensorBoard sessions, which can be either started from a notebook, or from a terminal (which is also available through the JupyterHub).

I guess the question should then be: should either of the two even _be_ a dependency? I'm no jupyter-server-proxy expert, I thought it piggy-backed of the JupyterHub proxy-ing? Not sure...

Reply:
I see, in that case it should be lowered to `IPython` then. That is the safest approach, it would work for your use case and also for those using JupyterLab without conflicts.

Reply:
What might be convenient is to make `JupyterHub` depend on `jupyter-server-proxy` since it is very common to use them together. So I suggest:
1. change dep on `jupyter-server-proxy` from JupyterLab to IPython
2. add optional dependency on `jupyter-server-proxy` to JupyterHub

Would this work for you?

Reply:
Works for me, I'll add it there.